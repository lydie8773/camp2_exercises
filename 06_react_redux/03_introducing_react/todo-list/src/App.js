import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import TodoForm from "./TodoForm";

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Todo list</h1>
        </header>
        <p className="App-intro">
					<div class="container">
						<TodoForm/>
					</div>
        </p>
      </div>
    );
  }
}

export default App;
