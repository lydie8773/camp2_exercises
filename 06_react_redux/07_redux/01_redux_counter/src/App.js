// import React, { Component } from 'react';
//
// import './App.css';
// import Counter from "./Counter";
// import { connect } from "react-redux";
//
//
// class App extends Component {
//
//   onIncrement() {
//     this.props.Increase();
//   }
//
//   onDecrement() {
//     this.props.Decrease();
//   }
//   render() {
//
//     return (
//       <div>
//         <h2>Counter</h2>
//         <p>The counter value is {this.props.count}</p>
//         <button onClick={this.onIncrement.bind(this)}>Increase</button>
//         <button onClick={this.onDecrement.bind(this)}>Decrease</button>
//       </div>
//     )
//   }
//
//
//
//   // render() {
//   //   return (
//   //     <div className="App">
//   //       <header className="App-header">
//   //         <img src={logo} className="App-logo" alt="logo" />
//   //         <h1 className="App-title">Welcome to React</h1>
//   //       </header>
//   //       <p className="App-intro">
//   //         {/*<Counter value={store.getState()}/>*/}
//   //         <Counter/>
//   //       </p>
//   //     </div>
//   //   );
//   // }
// }
//
// const mapStateToProps = function (state) {
//   return state;
// };
//
// const mapDispatchToProps = function (dispatch) {
//   return {
//     Increase: function () {
//       return dispatch({type: "COUNTER_INCRE"})
//     },
//     Decrease: function () {
//       return dispatch({type: "COUNTER_DECRE"})
//     }
//   }
// };
//
// export const CounterPackage = connect(mapStateToProps, mapDispatchToProps)(Counter);
//
//
// export default App;







import React, {Component} from 'react';
import {connect} from 'react-redux';

class App extends Component {

  onIncrement() {
    this.props.Increase();
  }

  onDecrement() {
    this.props.Decrease();
  }

  render() {

    console.log(this.props.count);
    return (
      <div>
        <h2>Counter</h2>
        <p>The counter value is {this.props.count}</p>
        <button onClick={this.onIncrement.bind(this)}>+</button>
        <button onClick={this.onDecrement.bind(this)}>-</button>
      </div>
    )
  }
}
const mapStateToProps = function (state) {
  return { count: state.count }
};

const mapDispatchToProps = function (dispatch) {
  return {
    Increase: function () {
      return dispatch({
        type: "COUNTER_INCRE"
      })
    },
    Decrease: function () {
      return dispatch({
        type: "COUNTER_DECRE"
      })
    }
  }
};

const ConnectedApp = connect(mapStateToProps, mapDispatchToProps)(App);


export default ConnectedApp;

