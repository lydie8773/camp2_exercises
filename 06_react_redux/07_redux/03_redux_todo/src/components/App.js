import React from 'react';

import TodoList from '../containers/TodoList';
import TodoInput from '../containers/TodoInput';

const App = () => {
  return(
    <div className="container">
      <h1 className="text-center">Your Todo List</h1>
      <TodoInput />
      <TodoList />

    </div>
  );
}

export default App;
