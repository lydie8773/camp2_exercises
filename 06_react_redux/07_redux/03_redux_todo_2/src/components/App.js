import React from 'react';

import Layout from './Layout';
import TodoList from '../containers/TodoList';
import TodoInput from '../containers/TodoInput';

const App = () => {
  return(
    <Layout>
      <TodoList />
      <TodoInput />
    </Layout>
  );
}

export default App;
