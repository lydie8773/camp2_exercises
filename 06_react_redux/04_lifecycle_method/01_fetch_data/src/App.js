import React, { Component } from 'react';
// import logo from './logo.svg';
import './App.css';





class DogApp extends Component {
	constructor(props) {
		super(props);
		this.state = {
			urlPic: ""
		};
		this.componentDidMount = this.componentDidMount.bind(this);
	}

	componentDidMount (){
		fetch("https://random.dog/woof.json", {
			method: "GET"
		})
			.then((response) => response.json())
			.then((data) => {
					console.log("data", data.url);
					this.setState({
						urlPic: data.url
					})
			})
	}


  render() {
    return (
      <div className="App">
				<button onClick={this.componentDidMount}>Change image</button>
				{
					this.state.urlPic ?
						(<img src={this.state.urlPic} alt="pic"></img>) : null
				}
      </div>
    );
  }
}

export default DogApp;
