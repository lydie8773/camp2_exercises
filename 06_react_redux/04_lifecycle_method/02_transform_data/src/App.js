import React, { Component } from 'react';
import './App.css';


// function getUserById(id, list){
// 	let userName;
// 	list.forEach((row) => {
// 		if(row.id === id){
// 			userName = row.name;
// 		}
// 	})
// 	return userName;
// }
//
// function Payment(props) {
// 	props.expenses.map((row, index) => {
// 		return (
// 			<table style="width:100%">
// 				<tr>
// 					<th>what has been paid</th>
// 					<th>by whom?</th>
// 				</tr>
// 				<tr key={index}>
// 					<td>{row.what}</td>
// 					<td>{this.getUserById(row.who)}</td>
// 				</tr>
// 			</table>
// 		)
// 	})
// }

// singleRowValue(row){
// 	return(
// 		Object.values(row).map((ele) => {
// 			return <td>{ele}</td>
// 		})
// 	);
// }
//
// tableValue(tableObject){
// 	return(
// 		tableObject.map((row) => {
// 			return <tr key={Object.values(row)[0]}>{this.singleRowValue(row)}</tr>
// 		})
// 	);
// }

// function Payment(props) {
// 	let userName;
//
// 	props.expenses.map((eleExpense) => {
// 		props.users.forEach((eleUsers) => {
// 			if (eleExpense.who === eleUsers.id){
// 				userName = eleUsers.name;
// 			}
// 		});
//
// 		return (
// 			<li>{eleExpense.what}, {userName}</li>
// 		)
// 	})
// }

// App has `expenses` and `users` as props
class App extends Component {
	// return (
	// 	<div className="App">
	// 		<Payment expenses={this.props.expenses} users={this.props.users} />
	// 	</div>
	// );

	// let userName;
	//
	// this.props.expenses.map((eleExpense) => {
	// 	this.props.users.forEach((eleUsers) => {
	// 		if (eleExpense.who === eleUsers.id) {
	// 			userName = eleUsers.name;
	// 		}
	// 	});
	//
	//
	// 	return (
	// 		<div className="App">
	// 			<li>{eleExpense.what}, {userName}</li>
	// 		</div>
	// 	)
	// })

	findUserById(id, list){
		let userName = " ";
		list.forEach((row) => {
			if(row.id === id){
				userName = row.name;
			}
		})
		return userName? userName:"Nobody";
	}

	payment(props){
		let listPayment = [];
		let json = {};
		props.expenses.forEach((ele) => {
			json = {"what": ele.what, "who": this.findUserById(ele.who, props.users)};
			listPayment.push(json);
		})
		return (
			listPayment.map((row, index) => {
				return (
					<tr key={index}>
						<td>{row.what}</td>
						<td>{row.who}</td>
					</tr>
				)
			})
		);
	}

	render(){
		return (
			<div className="container">
				<table className="table table-striped table-bordered table-responsive-md table-condensed datatable">
					<thead className="thead-dark">
						<tr>
							<th>what has been paid</th>
							<th>by whom?</th>
						</tr>
					</thead>
					<tbody>
						{this.payment(this.props)}
					</tbody>
				</table>

			</div>
		);
	}
}



export default App;
