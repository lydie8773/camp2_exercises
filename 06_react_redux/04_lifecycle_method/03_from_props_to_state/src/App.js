import React, { Component } from 'react';
import './App.css';
import Book from "./Book"

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isbn: "978059651774",
    }
  }
  handleChange = (event) => {
    this.setState({isbn: event.target.value});
  }
  handleSearch = () => {
		this.setState({isbn: this.state.isbn});
  }
  render() {
    return (
      <div className="App">
        <input type="text" value={this.state.isbn} onChange={this.handleChange} />
        <button onClick={this.handleSearch}>Search</button>
        <Book isbn={this.state.isbn} />
      </div>
    );
  }
}

export default App;
