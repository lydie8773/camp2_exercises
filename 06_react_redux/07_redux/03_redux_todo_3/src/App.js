import React, { Component } from 'react';
import './App.css';
import Todo from "./modules/Todo"

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">

        </header>
        <Todo />
      </div>
    );
  }
}


export default App;
