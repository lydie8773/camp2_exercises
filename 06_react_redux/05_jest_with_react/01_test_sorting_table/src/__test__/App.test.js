import React from 'react';
import renderer from 'react-test-renderer';
import App from "../App";
// import {products} from "../index";
import {shallow, configure} from "enzyme";


import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });


const products =
	[
		{ "decathlon_id": 8282689, "title": "Corne chasse 14cm", "price": 9.99 },
		{ "decathlon_id": 8354464, "title": "Basic L print Long Gold Fusion", "price": 9.99 },
		{ "decathlon_id": 8380024, "title": "RUN ELIOPRIME", "price": 54.99 },
		{ "decathlon_id": 8379970, "title": "Pantalon Gym", "price": 12.99 },
		{ "decathlon_id": 8247793, "title": "PALMES WADERS", "price": 24.99 },
		{ "decathlon_id": 8357549, "title": "MINIMIZER EDEN UNI  NOIR", "price": 19.99 },
		{ "decathlon_id": 8326155, "title": "Pantalon Training mesh marine", "price": 44.99 },
		{ "decathlon_id": 8329121, "title": "COUTEAU A PALOURDES", "price": 4.99 },
		{ "decathlon_id": 8370749, "title": "Doudoune Hike 100 garçon bleu", "price": 9.99 },
		{ "decathlon_id": 8298354, "title": "OREILLER CONFORT", "price": 6.99 },
		{ "decathlon_id": 8044622, "title": "2 guêtres RIDING noir", "price": 14.99 },
		{ "decathlon_id": 8249674, "title": "BOBINE FUN 2 3 4mm X 40 20 12m", "price": 6.99 },
		{ "decathlon_id": 8353265, "title": "Justaucorps manche longue Gym.", "price": 34.99 }
	]



test('Snapshot test: onClick()', () => {
	const component = renderer.create(<App lines={products}/>);

	let snapshot = component.toJSON();
	expect(snapshot).toMatchSnapshot();

	snapshot.children[0].children[0].children[0].props.onClick();
	snapshot = component.toJSON();
	expect(snapshot).toMatchSnapshot();

	snapshot.children[0].children[0].children[1].props.onClick();
	snapshot = component.toJSON();
	expect(snapshot).toMatchSnapshot();

	snapshot.children[0].children[0].children[2].props.onClick();
	snapshot = component.toJSON();
	expect(snapshot).toMatchSnapshot()
});

describe("Enzyme shallow", () => {
	it("Shallow test: onClick()", () => {
		const component = shallow(<App lines={products}/>);
		console.log("component: ", component.find("td").first().props());
		console.log("state: ", component.state().reverseSort);

		component.find("th").first().simulate("click");
		component.setState({reverseSort: false });
		console.log("state after: ", component.state().reverseSort);
		expect(component.find("td").first().props().children.toString()).toEqual("8044622");
	});
});

