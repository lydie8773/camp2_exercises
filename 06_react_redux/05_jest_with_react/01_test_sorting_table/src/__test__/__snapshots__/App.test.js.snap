// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test: onClick() 1`] = `
<table>
  <thead>
    <tr>
      <th
        onClick={[Function]}
      >
        ID
      </th>
      <th
        onClick={[Function]}
      >
        Title
      </th>
      <th
        onClick={[Function]}
      >
        Price
      </th>
    </tr>
  </thead>
  <tr>
    <td>
      8044622
    </td>
    <td>
      2 guêtres RIDING noir
    </td>
    <td>
      14.99
    </td>
  </tr>
  <tr>
    <td>
      8247793
    </td>
    <td>
      PALMES WADERS
    </td>
    <td>
      24.99
    </td>
  </tr>
  <tr>
    <td>
      8249674
    </td>
    <td>
      BOBINE FUN 2 3 4mm X 40 20 12m
    </td>
    <td>
      6.99
    </td>
  </tr>
  <tr>
    <td>
      8282689
    </td>
    <td>
      Corne chasse 14cm
    </td>
    <td>
      9.99
    </td>
  </tr>
  <tr>
    <td>
      8298354
    </td>
    <td>
      OREILLER CONFORT
    </td>
    <td>
      6.99
    </td>
  </tr>
  <tr>
    <td>
      8326155
    </td>
    <td>
      Pantalon Training mesh marine
    </td>
    <td>
      44.99
    </td>
  </tr>
  <tr>
    <td>
      8329121
    </td>
    <td>
      COUTEAU A PALOURDES
    </td>
    <td>
      4.99
    </td>
  </tr>
  <tr>
    <td>
      8353265
    </td>
    <td>
      Justaucorps manche longue Gym.
    </td>
    <td>
      34.99
    </td>
  </tr>
  <tr>
    <td>
      8354464
    </td>
    <td>
      Basic L print Long Gold Fusion
    </td>
    <td>
      9.99
    </td>
  </tr>
  <tr>
    <td>
      8357549
    </td>
    <td>
      MINIMIZER EDEN UNI  NOIR
    </td>
    <td>
      19.99
    </td>
  </tr>
  <tr>
    <td>
      8370749
    </td>
    <td>
      Doudoune Hike 100 garçon bleu
    </td>
    <td>
      9.99
    </td>
  </tr>
  <tr>
    <td>
      8379970
    </td>
    <td>
      Pantalon Gym
    </td>
    <td>
      12.99
    </td>
  </tr>
  <tr>
    <td>
      8380024
    </td>
    <td>
      RUN ELIOPRIME
    </td>
    <td>
      54.99
    </td>
  </tr>
</table>
`;

exports[`Snapshot test: onClick() 2`] = `
<table>
  <thead>
    <tr>
      <th
        onClick={[Function]}
      >
        ID
      </th>
      <th
        onClick={[Function]}
      >
        Title
      </th>
      <th
        onClick={[Function]}
      >
        Price
      </th>
    </tr>
  </thead>
  <tr>
    <td>
      8380024
    </td>
    <td>
      RUN ELIOPRIME
    </td>
    <td>
      54.99
    </td>
  </tr>
  <tr>
    <td>
      8379970
    </td>
    <td>
      Pantalon Gym
    </td>
    <td>
      12.99
    </td>
  </tr>
  <tr>
    <td>
      8370749
    </td>
    <td>
      Doudoune Hike 100 garçon bleu
    </td>
    <td>
      9.99
    </td>
  </tr>
  <tr>
    <td>
      8357549
    </td>
    <td>
      MINIMIZER EDEN UNI  NOIR
    </td>
    <td>
      19.99
    </td>
  </tr>
  <tr>
    <td>
      8354464
    </td>
    <td>
      Basic L print Long Gold Fusion
    </td>
    <td>
      9.99
    </td>
  </tr>
  <tr>
    <td>
      8353265
    </td>
    <td>
      Justaucorps manche longue Gym.
    </td>
    <td>
      34.99
    </td>
  </tr>
  <tr>
    <td>
      8329121
    </td>
    <td>
      COUTEAU A PALOURDES
    </td>
    <td>
      4.99
    </td>
  </tr>
  <tr>
    <td>
      8326155
    </td>
    <td>
      Pantalon Training mesh marine
    </td>
    <td>
      44.99
    </td>
  </tr>
  <tr>
    <td>
      8298354
    </td>
    <td>
      OREILLER CONFORT
    </td>
    <td>
      6.99
    </td>
  </tr>
  <tr>
    <td>
      8282689
    </td>
    <td>
      Corne chasse 14cm
    </td>
    <td>
      9.99
    </td>
  </tr>
  <tr>
    <td>
      8249674
    </td>
    <td>
      BOBINE FUN 2 3 4mm X 40 20 12m
    </td>
    <td>
      6.99
    </td>
  </tr>
  <tr>
    <td>
      8247793
    </td>
    <td>
      PALMES WADERS
    </td>
    <td>
      24.99
    </td>
  </tr>
  <tr>
    <td>
      8044622
    </td>
    <td>
      2 guêtres RIDING noir
    </td>
    <td>
      14.99
    </td>
  </tr>
</table>
`;

exports[`Snapshot test: onClick() 3`] = `
<table>
  <thead>
    <tr>
      <th
        onClick={[Function]}
      >
        ID
      </th>
      <th
        onClick={[Function]}
      >
        Title
      </th>
      <th
        onClick={[Function]}
      >
        Price
      </th>
    </tr>
  </thead>
  <tr>
    <td>
      8044622
    </td>
    <td>
      2 guêtres RIDING noir
    </td>
    <td>
      14.99
    </td>
  </tr>
  <tr>
    <td>
      8249674
    </td>
    <td>
      BOBINE FUN 2 3 4mm X 40 20 12m
    </td>
    <td>
      6.99
    </td>
  </tr>
  <tr>
    <td>
      8354464
    </td>
    <td>
      Basic L print Long Gold Fusion
    </td>
    <td>
      9.99
    </td>
  </tr>
  <tr>
    <td>
      8329121
    </td>
    <td>
      COUTEAU A PALOURDES
    </td>
    <td>
      4.99
    </td>
  </tr>
  <tr>
    <td>
      8282689
    </td>
    <td>
      Corne chasse 14cm
    </td>
    <td>
      9.99
    </td>
  </tr>
  <tr>
    <td>
      8370749
    </td>
    <td>
      Doudoune Hike 100 garçon bleu
    </td>
    <td>
      9.99
    </td>
  </tr>
  <tr>
    <td>
      8353265
    </td>
    <td>
      Justaucorps manche longue Gym.
    </td>
    <td>
      34.99
    </td>
  </tr>
  <tr>
    <td>
      8357549
    </td>
    <td>
      MINIMIZER EDEN UNI  NOIR
    </td>
    <td>
      19.99
    </td>
  </tr>
  <tr>
    <td>
      8298354
    </td>
    <td>
      OREILLER CONFORT
    </td>
    <td>
      6.99
    </td>
  </tr>
  <tr>
    <td>
      8247793
    </td>
    <td>
      PALMES WADERS
    </td>
    <td>
      24.99
    </td>
  </tr>
  <tr>
    <td>
      8379970
    </td>
    <td>
      Pantalon Gym
    </td>
    <td>
      12.99
    </td>
  </tr>
  <tr>
    <td>
      8326155
    </td>
    <td>
      Pantalon Training mesh marine
    </td>
    <td>
      44.99
    </td>
  </tr>
  <tr>
    <td>
      8380024
    </td>
    <td>
      RUN ELIOPRIME
    </td>
    <td>
      54.99
    </td>
  </tr>
</table>
`;

exports[`Snapshot test: onClick() 4`] = `
<table>
  <thead>
    <tr>
      <th
        onClick={[Function]}
      >
        ID
      </th>
      <th
        onClick={[Function]}
      >
        Title
      </th>
      <th
        onClick={[Function]}
      >
        Price
      </th>
    </tr>
  </thead>
  <tr>
    <td>
      8329121
    </td>
    <td>
      COUTEAU A PALOURDES
    </td>
    <td>
      4.99
    </td>
  </tr>
  <tr>
    <td>
      8298354
    </td>
    <td>
      OREILLER CONFORT
    </td>
    <td>
      6.99
    </td>
  </tr>
  <tr>
    <td>
      8249674
    </td>
    <td>
      BOBINE FUN 2 3 4mm X 40 20 12m
    </td>
    <td>
      6.99
    </td>
  </tr>
  <tr>
    <td>
      8282689
    </td>
    <td>
      Corne chasse 14cm
    </td>
    <td>
      9.99
    </td>
  </tr>
  <tr>
    <td>
      8354464
    </td>
    <td>
      Basic L print Long Gold Fusion
    </td>
    <td>
      9.99
    </td>
  </tr>
  <tr>
    <td>
      8370749
    </td>
    <td>
      Doudoune Hike 100 garçon bleu
    </td>
    <td>
      9.99
    </td>
  </tr>
  <tr>
    <td>
      8379970
    </td>
    <td>
      Pantalon Gym
    </td>
    <td>
      12.99
    </td>
  </tr>
  <tr>
    <td>
      8044622
    </td>
    <td>
      2 guêtres RIDING noir
    </td>
    <td>
      14.99
    </td>
  </tr>
  <tr>
    <td>
      8357549
    </td>
    <td>
      MINIMIZER EDEN UNI  NOIR
    </td>
    <td>
      19.99
    </td>
  </tr>
  <tr>
    <td>
      8247793
    </td>
    <td>
      PALMES WADERS
    </td>
    <td>
      24.99
    </td>
  </tr>
  <tr>
    <td>
      8353265
    </td>
    <td>
      Justaucorps manche longue Gym.
    </td>
    <td>
      34.99
    </td>
  </tr>
  <tr>
    <td>
      8326155
    </td>
    <td>
      Pantalon Training mesh marine
    </td>
    <td>
      44.99
    </td>
  </tr>
  <tr>
    <td>
      8380024
    </td>
    <td>
      RUN ELIOPRIME
    </td>
    <td>
      54.99
    </td>
  </tr>
</table>
`;
